resource_activity/models/resource_activity_registration.py

```diff
@@ -159,6 +92,73 @@ def _compute_state(self):
     )
     is_accessory_registration = fields.Boolean(related="resource_category.is_accessory")
 
+    @api.onchange("resource_category")
+    def onchange_resource_category(self):
+        self.product_id = False
+        if self.is_accessory_registration:
+            self.quantity = 0
+
+    @api.onchange("quantity")
+    def onchange_quantity(self):
+        if not self.bring_bike and self.state not in ["option", "booked"]:
+            self.quantity_needed = self.quantity
+
+    @api.onchange("quantity_needed")
+    def onchange_quantity_needed(self):
+        if self.state not in ["booked", "option", "draft"]:
+            if self.quantity_needed > self.quantity_allocated:
+                self.state = "waiting"
+
+    @api.onchange("bring_bike")
+    def onchange_bring_bike(self):
+        if self.bring_bike:
+            self.quantity_needed = 0
+        else:
+            self.quantity_needed = self.quantity
+
+    @api.onchange("booking_type")
+    def onchange_booking_type(self):
+        if self.booking_type == "booked":
+            self.date_lock = None
+
+    @api.multi
+    @api.depends("quantity_needed", "quantity", "product_id", "state")
+    def _compute_need_push(self):
+        for registration in self:
+            # if sale order was created for registration, any
+            # any change must be pushed
+            if registration.sale_order_id:
+                registration.need_push = True
+
+            # if new registrations goes to booked or option state
+            elif (
+                registration.resource_activity_id.sale_orders
+                and registration.state in ["booked", "option"]
+            ):
+                registration.need_push = True
+
+    @api.multi
+    @api.depends("allocations.state")
```

Comment:
```suggestion
    @api.depends("allocations", "allocations.state")
```
like this, it would also work if an allocation is deleted.